shared/utils/logic.ts

```diff
@@ -265,102 +312,137 @@ export const getVisibleFieldIds = (
  * @returns true if all the conditions are satisfied, false otherwise
  */
 const isLogicUnitSatisfied = (
-  submission: LogicFieldSchemaOrResponse[],
-  logicUnit: IConditionSchema[],
+  submission: LogicFieldResponse[],
+  logicUnit: FormCondition[],
   visibleFieldIds: FieldIdSet,
-): boolean => {
-  return logicUnit.every((condition) => {
+): boolean =>
+  logicUnit.every((condition) => {
     const conditionField = findConditionField(submission, condition.field)
     return (
       conditionField &&
       visibleFieldIds.has(conditionField._id.toString()) &&
       isConditionFulfilled(conditionField, condition)
     )
   })
-}
 
 const getCurrentValue = (
-  field: LogicFieldSchemaOrResponse,
-): string | null | undefined | string[] => {
-  if ('fieldValue' in field) {
+  field: LogicFieldResponse,
+): string | LogicFieldClientRadioResponseInput | null | undefined => {
+  if ('input' in field) {
     // client
-    return field.fieldValue
+    return field.input
   } else if ('answer' in field) {
     // server
     return field.answer
   }
   return null
 }
+
 /**
  * Checks if the field's value matches the condition
  * @param {Object} field
  * @param {Object} condition
  * @param {String} condition.state - The type of condition
  */
 const isConditionFulfilled = (
-  field: LogicFieldSchemaOrResponse,
-  condition: IConditionSchema,
+  field: LogicFieldResponse,
+  condition: FormCondition,
 ): boolean => {
-  if (!field || !condition) {
-    return false
-  }
   let currentValue = getCurrentValue(field)
+
+  // Trim the current value
+  if (typeof currentValue === 'string') {
+    currentValue = currentValue.trim()
+  }
+
   if (
     currentValue === null ||
     currentValue === undefined ||
```

Comment:
/nit + not part of your code change

Can be terser with
```suggestion
    currentValue == null
```